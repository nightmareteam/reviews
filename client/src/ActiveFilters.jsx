import React from 'react';
//import styled from 'styled-components';
const styled = window.styled;

/**
 * This component renders the active filters
 * @param {Object} activeFilters - a filter object with active filters { id, options: {optionId, optionName}}
 * @param {Function} setFilters - a function to update the selected filters
 */
const ActiveFilters = ({activeFilters, setFilters}) => (
  <FilterWrapper>
    {'Filters'}
    {
      Object.keys(activeFilters).map(key => (
        activeFilters[key] ? 
          (
            <FilterButton key={key} onClick={(e) => setFilters(e, key, {optionId: {}, optionId: {}})}>
              <FilterName >{activeFilters[key].optionName}</FilterName>
            </FilterButton>
          )
          : null
      ))
    }
  </FilterWrapper>
);

const FilterWrapper = styled.div`
  padding-top: 10px;
  padding-bottom: 0px;
  color: #c6d4df;
  font-size: 15px;
  display: inline-block;
  text-align: center;
`;

const FilterButton = styled.div`
  border-radius: 2px;
  padding-right: 25px;
  padding-left: 5px;
  background: #485260;
  display: inline-block;
  font-size: 12px;
  height: 24px;
  margin-left: 10px;
  vertical-align: middle;
  line-height: 24px;
  background-image: url('/images/deleteSearchTerm.png');
  background-repeat: no-repeat;
  background-position: right 5px center;
  cursor: pointer;
`;

const FilterName = styled.div``;

export default ActiveFilters;