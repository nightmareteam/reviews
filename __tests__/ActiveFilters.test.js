import React from 'react';
import { shallow } from 'enzyme';
import ActiveFilters from '../client/src/ActiveFilters.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({
  activeFilters: {
    language: {
      optionId: 'arabic',
      optionName: 'Arabic'
    }
  },
  setFilters: mockCallback
});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<ActiveFilters activeFilters={props.activeFilters} setFilters={mockCallback} /> );
});

describe('<ActiveFilters /> rendering', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});