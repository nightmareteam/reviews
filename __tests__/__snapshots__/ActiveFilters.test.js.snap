// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveFilters /> rendering should render elements 1`] = `
<styled.div>
  Filters
  <styled.div
    key="language"
    onClick={[Function]}
  >
    <styled.div>
      Arabic
    </styled.div>
  </styled.div>
</styled.div>
`;
